### PR TITLE
Added option B for blob support

### PR DESCRIPTION
**Pull request by mdiazjusticia**
Option attributeValue.getB for loading binary data from DynamoDB

### CODE REVIEW

**Review by avram — CHANGES_REQUESTED** [review]
Can you add this to the tests? Also wondering how you're representing binary content in the JSON.

**Review by casidiablo — CHANGES_REQUESTED** [review]
I'm not sure it will "_just work_". This covers the deserialization of records... but I'm especially intrigued about the serialization counterpart.

Can you please add a test to exercise this new capability?

File: src/main/java/com/scopely/mapper/JsonNodeAttributeValueMapper.java [hunk 1]
```diff
@@ -110,6 +114,12 @@ public static ObjectNode convert(Map<String, AttributeValue> map, ObjectMapper o
                 return setAVForArray(node, attributeValue);
             case MISSING:
             case BINARY:
+                try {
```
Comment (avram) [inline src/main/java/com/scopely/mapper/JsonNodeAttributeValueMapper.java]: This doesn't look right. Surely you don't mean to use this logic for the MISSING case.

**Review by avram — CHANGES_REQUESTED** [review]
In addition to my comment, can you make sure that you have a test case that uses a model object that you are round tripping to Dynamo?

**Review by casidiablo — APPROVED** [review]

**Review by avram — COMMENTED** [review]
On phone with bad internet, but two non blocking notes: Include cause when wrapping the exception as MappingException and always specify a charset when calling getBytes on a string. Even in tests- treat the version with default charset as deprecated.

File: src/test/java/com/scopely/mapper/JsonNodeAttributeValueMapperTest.java [hunk 2]
```diff
@@ -24,6 +26,24 @@ public void convertToMap(String json, Map<String, Set<Map.Entry<String, Attribut
         assertThat(attributeValueMap).isEqualTo(map);
     }
 
+    @Test
+    public void convertBinaryToMap() throws Exception {
+        ObjectNode node = new ObjectMapper().createObjectNode();
+        String base64BinaryInfo = "12345=";
+        node.put("key", ByteBuffer.wrap(base64BinaryInfo.getBytes("UTF-8")).array());
```
Comment (avram) [inline src/test/java/com/scopely/mapper/JsonNodeAttributeValueMapperTest.java:33]: StandardCharsets.UTF8 or similar

File: src/test/java/com/scopely/mapper/TablePersistenceTest.java [hunk 3]
```diff
@@ -500,6 +501,29 @@ public void query_all() throws Exception {
...
+    @Test
+    public void simple_free_built_binary_persists() throws Exception {
+        dynamoLocal.createTable(ctr -> {
+            ctr.setTableName("simple_free_built_with_binary");
+            ctr.setKeySchema(ImmutableList.of(new KeySchemaElement("hashKey", KeyType.HASH)));
+            ctr.setAttributeDefinitions(ImmutableList.of(new AttributeDefinition("hashKey", ScalarAttributeType.S)));
+        });
+
+        JsonDynamoMapper jsonDynamoMapper = new JsonDynamoMapper(amazonDynamoDBClient);
+        ByteBuffer bb = ByteBuffer.wrap("val".getBytes());
```
Comment (avram) [inline src/test/java/com/scopely/mapper/TablePersistenceTest.java:513]: Here too

**Review by avram — APPROVED** [review]

**Pull request merged**